src/status_im/chat/events/receive_message.cljs

```diff
@@ -60,10 +60,13 @@
              (not= from (:public-key current-account))
              (pop-up-chat? chat-identifier))
       (let [group-chat?      (not (nil? group-id))
+            chat-exists?     (get-in db [:chats chat-identifier])
+            fx               (if chat-exists?
+                               (model/upsert-chat cofx {:chat-id    chat-identifier
```

Comment:
Does this mean this is really `update-chat`? If it is upsert it should deal with chat not existing, right?

Reply:
that's not what the function does and it's used elsewhere as well, the point of this PR was just to fix the bug with no impact on the rest of the code

Reply:
Well then that's still a bug. The function name should reflect what it does. If it doesn't upsert, then I'd rather we add a copy function `update` and add a TODO to `upsert-chat` that says "XXX: This function doesn't upsert!". That or encapsulate the if statement in the proper place, which is fixing the upsert so it actually upserts.

Reply:
We also call upsert-chat in many places, so if this is what other functions think this function does then that's a systemic problem. We should make sure to at least capture this in all the places it is used if we don't have time to fix it, otherwise we are just going to have to have 5 (just grep, not sure if this is actual code paths) more PRs that does essentially same thing.